### ISSUE TITLE
Find All References on overridden members is too broad

### ISSUE THREAD

**Issue by Spongman** [dotnet/roslyn#40723]
**Version Used**: vs 16.4.1

**Steps to Reproduce**:
````
public interface IFoo
{
	void Foo() { }
}
public interface IBar
{
	void Bar() { }
}
public class Base : IFoo, IBar
{
	public virtual void Foo() { }
	public virtual void Bar() { }
	public virtual void Goo() { }
}
public class Derived : Base, IFoo
{
	public override void Foo() { }	// 'Find All References' on 'Foo'
	public override void Bar() { }  // 'Find All References' on 'Bar'
	public override void Goo() { }  // 'Find All References' on 'Goo'
}
public class Other : Base, IFoo
{
	public override void Foo() { } 
	public override void Bar() { } 
	public override void Goo() { }
}
public class Program
{
	public static void Main()
	{
		new Other().Foo();
		new Other().Bar();
		new Other().Goo();
	}
}
````

do the 'Find All References' on `Derived.Foo|Bar|Goo`. notice that results are returned for `Other.Foo|Bar|Goo` also.

This is fine in the above example, but if you have a base class from which 1000s of classes are derived, this makes this operation essentially useless.

The problem is: there is already a way to 'Find All References' for `Base` members - go to `Base` and do the search from there. However, there's NO WAY to _only_ search for `Derived` members.

Instead of having two ways to do one thing, and no ways to do the other. please make one way to do each thing.

**Comment by CyrusNajmabadi** [comment]
The results should support grouping (and that should be the default).  So you should see:

![image](https://user-images.githubusercontent.com/4564579/71750159-78e60280-2e2c-11ea-9600-032e024b3c8e.png)

All the direct references to `Other.Whatever` should be grouped underneath it.  You can use the different grouping options to enable this:

![image](https://user-images.githubusercontent.com/4564579/71750203-92874a00-2e2c-11ea-93a8-30ae64963618.png)

**Comment by stormCup** [comment]
Yeah the grouping doesn’t help. I still get a HUGE list of irrelevant search results that I could have gotten just fine by going to the base class.
I want to see the references to the member I clicked on, not all the references to all other implementations of that member.
Please make it not suck.

**Comment by sharwell** [comment]
@stormCup You should be able to use the Call Hierarchy view for this. When you right click the method, it's the option immediately below Find All References.

**Comment by Spongman** [comment]
this issue isn't about Call Hierarchy, it's about 'Find All References'. Using Call Hierarchy isn't even a good workaround in the use-case given, where there are many derived classes that implement the base-defined method. both result in an unusable interface that requires a ton of manual searching through a list of irrelevant results. 

the whole point of having a search function is to find the things you're looking for. if that search function returns 99.9% irrelevant results, it's not doing its job.

**Comment by CyrusNajmabadi** [comment]
>  if that search function returns 99.9% irrelevant results, it's not doing its job.

the question is if they're relevant to some people.  I know they're relevant to me almost all of the time.  In the cases where they're not, i just examine the references under the specific item (or items) i care about.

We could potentially have a button to control 'cascading here'.  I would not be opposed to that.  Not sure what hte priority would be on something like that though given there is only one report of an issue here.

**Comment by jrmoreno1** [comment]
It's definitely a pain point for me that I have just been living with (along with FAR for a method on a generic interface).

**Comment by sharwell** [comment]
This should be substantially improved by #51637

**Comment by jrmoreno1** [comment]
> This should be substantially improved in 16.10 Preview 2 by #51637

TL;DR Thanks.

I don’t have preview 10 installed, but reading the description, it at least sounds like what I would want.  If I have b.Foo (abstract) and d1.Foo, and d2.Foo and d99.Foo, and I try to find all references to d99.Foo, I don’t care about the other references, and as @Spongman said, if I did care, it would be easy to go to base and do the find all from there.  From reading #51637 it sounds like that is mainly what it would do.  

I think this will be a tremendous improvement once it gets to a version I can use.
